src/iop/demosaic.c

```diff
@@ -5769,9 +5769,10 @@ void gui_focus(struct dt_iop_module_t *self, gboolean in)
   dt_iop_demosaic_gui_data_t *g = (dt_iop_demosaic_gui_data_t *)self->gui_data;
   if(!in)
   {
+    const gboolean was_dualmask = dt_bauhaus_widget_get_quad_active(g->dual_thrs);
```

Comment:
it looks like this could be simplified using the recorded boolean state:

`const gboolean mask_visible = g->visual_mask;`

Reply:
absolutely right you are :-)